From e0a8d2005c68a4b2ca1543149f50c03c3c7c0b5f Mon Sep 17 00:00:00 2001
From: Olivier Fourdan <ofourdan@redhat.com>
Date: Tue, 4 Mar 2025 16:21:30 +0100
Subject: [PATCH 1/2] search-app: Add systemd session files

The search appliance session would fail to start as a systemd session.

Add the required systemd plumbing to fix the search appliance session.

(cherry picked from commit 5308432e1c564ec096acbb3f4fb19f940d6c2ae9)
---
 meson.build                                          | 12 ++++++++++++
 .../systemd/org.gnome.Kiosk.SearchApp.service.in     |  8 ++++++++
 search-app/systemd/session.conf                      |  3 +++
 3 files changed, 23 insertions(+)
 create mode 100644 search-app/systemd/org.gnome.Kiosk.SearchApp.service.in
 create mode 100644 search-app/systemd/session.conf

diff --git a/meson.build b/meson.build
index 0766f8d..e3d8ff3 100644
--- a/meson.build
+++ b/meson.build
@@ -188,3 +188,15 @@ i18n.merge_file('desktop',
         install_dir: desktop_data_dir,
         type: 'desktop'
 )
+
+configure_file(
+       input: 'search-app/systemd/org.gnome.Kiosk.SearchApp.service.in',
+       output: '@BASENAME@',
+       configuration: systemd_service_config_data,
+       install_dir: systemd_user_unit_dir
+)
+
+kiosk_search_appliance_systemd_target_dir = join_paths(systemd_user_unit_dir, 'gnome-session@org.gnome.Kiosk.SearchApp.target.d')
+install_data('search-app/systemd/session.conf',
+        install_dir: kiosk_search_appliance_systemd_target_dir
+)
diff --git a/search-app/systemd/org.gnome.Kiosk.SearchApp.service.in b/search-app/systemd/org.gnome.Kiosk.SearchApp.service.in
new file mode 100644
index 0000000..29ddcd8
--- /dev/null
+++ b/search-app/systemd/org.gnome.Kiosk.SearchApp.service.in
@@ -0,0 +1,8 @@
+[Unit]
+Description=Kiosk Search Appliance
+BindsTo=gnome-session.target
+After=gnome-session.target
+
+[Service]
+ExecStart=@bindir@/firefox --kiosk --private-window --new-instance  https://www.google.com
+Restart=always
diff --git a/search-app/systemd/session.conf b/search-app/systemd/session.conf
new file mode 100644
index 0000000..1ab41da
--- /dev/null
+++ b/search-app/systemd/session.conf
@@ -0,0 +1,3 @@
+[Unit]
+Requires=org.gnome.Kiosk.target
+Requires=org.gnome.Kiosk.SearchApp.service
-- 
2.48.1

